plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/worker/SerializableRuleSet.kt

```diff
@@ -0,0 +1,51 @@
+package org.jlleitschuh.gradle.ktlint.worker
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.RuleSet
+import java.io.IOException
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.io.Serializable
+import kotlin.reflect.jvm.jvmName
+
+/**
+ * Special wrapper around [RuleSet] allowing to serialize/deserialize it into file.
+ *
+ * Should be removed once KtLint will add interface implementation into [RuleSet].
+ */
+internal class SerializableRuleSet(
+    ruleSet: RuleSet
+) : Serializable {
+    @Transient
+    var ruleSet: RuleSet = ruleSet
+        private set
+
+    @Throws(IOException::class)
+    private fun writeObject(out: ObjectOutputStream) {
+        out.writeUTF(ruleSet.id)
+        out.writeObject(
+            ruleSet.rules.map { it::class.jvmName }
+        )
+    }
+
+    @Throws(IOException::class, ClassNotFoundException::class)
+    private fun readObject(oin: ObjectInputStream) {
+        val id = oin.readUTF()
+        @Suppress("UNCHECKED_CAST")
+        val rulesMaps = oin.readObject() as List<String>
+        val classLoader = this.javaClass.classLoader
+        val rules = rulesMaps
+            .map {
+                val ruleClass = classLoader.loadClass(it)
+                ruleClass.newInstance() as Rule
+            }
+        ruleSet = RuleSet(
+            id,
+            *rules.toTypedArray()
+        )
+    }
+
+    companion object {
+        private const val serialVersionUID: Long = 2012054L
+    }
+}
```

Comment:
Not going to lie, seeing hand-crafted object deserialization is making me cringe a bit.

This seems like the perfect place for us to introduce issues when ktlint releases API changing updates.

Reply:
Yes, I plan to add `Serializable` to this objects in next KtLint release. For now we have to use this wrappers.